sdk/react/src/SwapComponents.tsx

```diff
@@ -0,0 +1,22 @@
+import { ComponentProps, ComponentType } from 'react'
+import useVariableValue from './useVariableValue'
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export const SwapComponents = <T extends ComponentType<any>>(
```

Comment:
Swap components makes me almost feel like this could swap between `n` number of different components. Maybe we could use the name of the component to match against a string value?

Reply:
there's definitely a more complicated version  we could have where we map the possible values of variable to the component that should be rendered.